### PR TITLE
Shifting LaTeX Code and Draw API Refactor

### PR DESCRIPTION
**Pull request by gadhvirushiraj**
## Shifting LaTeX Code and Circuit API Refactor

Transferring all LaTeX circuit rendering functionality from `circuit.py` to a dedicated file named `texrenderer.py`. Additional renderers such as `matplotlib` and `text` will be incorporated into the `draw()` method as they become available.

### CODE REVIEW

**Comment by gadhvirushiraj** [comment]
@BoxiLi  should we move `circuit_latex.py` as well, or maybe rename it to `latex_utils.py` ?

**Comment by BoxiLi** [comment]
I think it would be nice to merge them into `texrenderer.py`. One file suffice, it is not so big.

**Comment by BoxiLi** [comment]
I activated the tests but some of them are failing I believe. You can verify by `pytest tests` locally on your machine.

**Comment by gadhvirushiraj** [comment]
@BoxiLi All tests are passing, and `circuit_latex.py` has been moved. Do we need to include any particular deprecation message, considering `qc.draw()` continues to function as it did before?

File: src/qutip_qip/circuit/circuit.py [hunk 1]
```diff
@@ -1119,17 +997,19 @@ def draw(
...
+            else:
+                mode = "wb"
+                if file_type == "png" and not dpi:
+                    dpi = 100
+            latex = TeXRenderer(self)
+            image_data = latex.raw_img(file_type, dpi)
+            with open(
+                os.path.join(file_path, file_name + "." + file_type), mode
+            ) as f:
+                f.write(image_data)
```
Comment (BoxiLi) [inline src/qutip_qip/circuit/circuit.py:1012]: Please add a else statement here and an raise an error saying something like `f"Unknown {renderer}, please choose from "latex_render", ... and ...` etc. Right now there is only latex rendering of course.

**Review by BoxiLi — CHANGES_REQUESTED** [review]
Left one comment, will merge after this is addressed.

File: src/qutip_qip/circuit/circuit.py (hunk 1 above)
Comment (gadhvirushiraj) [reply src/qutip_qip/circuit/circuit.py:1012]: Added the error message and kept the options simple: latex, matplotlib, and text for easy recall.

**Review by BoxiLi — APPROVED** [review]

**Pull request merged**